### PR TITLE
Battle Overlay Progress + Documentation

### PR DESCRIPTION
**Pull request by adrienntindall**

### CODE REVIEW

File: src/battle/overlay_12_0224E4FC.c [hunk 1]
```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+    if (BattlerCheckSubstitute(ctx, ctx->battlerIdTarget) == TRUE) {
+        return ret;
+    }
+    
+    switch (GetBattlerAbility(ctx, ctx->battlerIdTarget)) {
+    case ABILITY_STATIC:
+        if (ctx->battleMons[ctx->battlerIdAttacker].hp &&
+            !ctx->battleMons[ctx->battlerIdAttacker].status &&
+            !(ctx->moveStatusFlag & MOVE_STATUS_FAIL) &&
+            !(ctx->linkStatus & 0x20) &&
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:3907]: use shifts

File: src/battle/overlay_12_0224E4FC.c [hunk 2]
```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+            !(ctx->linkStatus2 & 0x10) &&
+            (ctx->selfTurnData[ctx->battlerIdTarget].unk4 || ctx->selfTurnData[ctx->battlerIdTarget].unkC) &&
+            (ctx->unk_334.moveData[ctx->moveNoCur].unkB & 1) &&
+            (BattleSystem_Random(bsys) % 10 < 3)) {
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:3911]: I'd prefer brackets to make the order of ops clear and explicit

File: src/battle/overlay_12_0224E4FC.c [hunk 3]
```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+        if (ctx->battleMons[battlerId].item) {
+            ctx->battleMons[battlerId].unk88.knockOffFlag = TRUE;
+        }
+        break;
+    }
+    if (ret == TRUE) {
+        ctx->battlerIdWork = battlerId;
+        ctx->abilityWork = GetBattlerAbility(ctx, battlerId);
+        if (!flag) {
+            ReadBattleScriptFromNarc(ctx, NARC_a_0_0_1, 221);
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:4111]: no naix for the narc?

File: src/battle/overlay_12_0224E4FC.c [hunk 4]
```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+                    script = 206;
+                }
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_HP_RESTORE_SPICY: //figy berry
+            if (ctx->battleMons[battlerId].hp <= ctx->battleMons[battlerId].maxHp / 2) {
+                ctx->hpCalcWork = DamageDivide(ctx->battleMons[battlerId].maxHp, boost);
+                ctx->msgWork = 0;
+                if (GetFlavorPreferenceFromPID(ctx->battleMons[battlerId].personality, 0) == -1) {
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:4332]: personality enum/consts?

**Review by red031000 — CHANGES_REQUESTED** [review]
also can we delete work

File: src/battle/overlay_12_0224E4FC.c (hunk 3 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:4111]: Nope, we haven't decompiled the battle script narc yet for whatever reason

File: src/battle/overlay_12_0224E4FC.c (hunk 4 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:4332]: huh? you mean the 0 in the function?

Comment (red031000) [reply src/battle/overlay_12_0224E4FC.c:4332]: yeah, that varies with each personality difference

Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:4332]: oh I think those are flavor consts, let me see if we have them and I'll add them if not

File: src/battle/overlay_12_0224E4FC.c (hunk 1 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:3907]: there will be shifts once I go back through and put in constants lol

**Review by red031000 — APPROVED** [review]

**Pull request merged**